src/world/StatusEffect/StatusEffect.cpp

```diff
@@ -55,16 +68,53 @@ void Sapphire::StatusEffect::StatusEffect::registerTickEffect( uint8_t type, uin
 
 std::pair< uint8_t, uint32_t > Sapphire::StatusEffect::StatusEffect::getTickEffect()
 {
-  auto thisTick = m_currTickEffect;
-  m_currTickEffect = std::make_pair( 0, 0 );
-  return thisTick;
+  switch( static_cast< Common::StatusEffectType >( m_effectEntry.effectType ) )
+  {
+    case Common::StatusEffectType::Dot:
+    {
+      auto value = m_value;
+      if( m_cachedSourceCrit > Sapphire::Math::CalcStats::getRandomNumber0To99() )
+      {
+        value *= m_cachedSourceCritBonus;
+      }
+      value *= 1.0f + ( ( Sapphire::Math::CalcStats::getRandomNumber0To99() - 50.0f ) / 1000.0f );
+      m_currTickEffect = std::make_pair( 1, value );
+      break;
+    }
+
+    case Common::StatusEffectType::Hot:
+    {
+      auto value = m_value;
+      if( m_cachedSourceCrit > Sapphire::Math::CalcStats::getRandomNumber0To99() )
+      {
+        value *= m_cachedSourceCritBonus;
+      }
+      value *= 1.0f + ( ( Sapphire::Math::CalcStats::getRandomNumber0To99() - 50.0f ) / 1000.0f );
+      m_currTickEffect = std::make_pair( 2, value );
```

Comment:
use enums instead of literals, not sure what is with the obsession of using literals - it's really lazy and obscure

Reply:
this is not added by me. I just return the value needed for it to work.